src/WebPush.php

```diff
@@ -254,8 +254,8 @@ private function prepare(array $notifications): array
                 $headers['Topic'] = $options['topic'];
             }
 
-            // if GCM
-            if (substr($endpoint, 0, strlen(self::GCM_URL)) === self::GCM_URL) {
+            // if GCM or FCM using legacy GCM server key
+            if (array_key_exists('GCM', $auth) || substr($endpoint, 0, strlen(self::GCM_URL)) === self::GCM_URL) {
```

Comment:
This is breaking for people who use VAPID for new users, but still GCM for old ones. This should be an additional parameter